packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -836,14 +839,26 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         private _storage?: IDocumentStorageService,
     ) {
         super();
-
+        this.metadata = metadata ?? this.formMetadata();
```

Comment:
It appears to me (but I'm no expert) that `formMetadata()` is not "ready" to be called right at the very top here.  e.g. `this.disableIsolatedChannels` is read in there, but it is modified here in the constructor below.

`formMetadata` seems to be a helper for writing out the metadata blob, I wonder if you can just avoid using it here at all.  On a related note, the object it returns should probably account for the stuff you've set on `this.metadata`...

Here's one approach to consider:
* Change `this.metadata`'s type to be just the subset of props that you added, that are for tracking. There's a few ways to do this, lmk if you want to discuss.
* Also move its initialization down to the green block below where you're setting the values anyway.
* Also rename it, here I'll call it `this.metadataSiAdded` (there, now you have to come up with a reasonable name yourself 😉)
* Update `formMetadata` to include `...this.metadataSiAdded

Reply:
+1 to this. I would only add properties that are logged / tracked on be part of `this.metadata` (or whatever the new name is).
One way is that you can declare a new interface `IContainerRuntimeCreateMetadata` with the properties you are interested in. `this.metdata` becomes of this type. `IContainerRuntimeMetadata` can then extend `IContainerRuntimeCreateMetadata`.

Reply:
thank you both!